buildkitereporter/buildkitereporter.go

```diff
@@ -0,0 +1,168 @@
+package buildkitereporter
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/samsarahq/taskrunner"
+)
+
+type ReporterOption func(r *reporter)
+
+// WithPath sets a save path for the buildkite annotation (default: "taskrunner.annotation.md").
+func WithPath(path string) ReporterOption {
+	return func(r *reporter) { r.path = path }
+}
+
+// WithMaxLines sets the maximum stderr lines per task in the annotation (default: 50).
+func WithMaxLines(maxLines int) ReporterOption {
+	return func(r *reporter) { r.maxLines = maxLines }
+}
+
+// WithUpload is mutually exclusive vs WithMaxLines. Uploads the buildkite annotation directly.
```

Comment:
why is it mutually exclusive with maxlines? should this be `WithPath`?

Reply:
That's what I meant, whoops